components/offer/MakeOffer.vue

```diff
@@ -0,0 +1,241 @@
+const {
+  transaction,
+  isLoading,
+  status,
+  isError,
+  blockNumber,
+  txHash,
+  clear: clearTransaction,
+} = useTransaction()
+
+const { isTransactionSuccessful } = useTransactionSuccessful({
+  status,
+  isError,
+})
+
+const { decimals } = useChain()
+const { $i18n } = useNuxtApp()
+
+const autoTeleport = ref(false)
+const autoTeleportButton = ref()
+const itemCount = ref(offerStore.count)
+const items = ref<MakingOfferItem[]>([])
+const autoTeleportLoaded = ref(false)
+
+const isSuccessModalOpen = computed(
+  () => Boolean(items.value.length) && isTransactionSuccessful.value,
+)
+
+const getAction = (items: MakingOfferItem[]): Actions => {
+  return {
+    interaction: ShoppingActions.MAKE_OFFER,
+    urlPrefix: urlPrefix.value,
+    token: items.map(item => ({
+      price: String(calculateBalance(Number(item.offerPrice), decimals.value)),
+      collectionId: item.collection.id,
+      nftSn: item.sn,
+    } as TokenToOffer)),
+    duration: 300,
+    successMessage: $i18n.t('transaction.price.offer') as string,
+    errorMessage: $i18n.t('transaction.price.error') as string,
+  }
+}
+
+const { action } = useAutoTeleportActionButton({
+  autoTeleport,
+  autoTeleportButton,
+  autoTeleportLoaded,
+  getActionFn: () => getAction(offerStore.itemsInChain),
+})
+
+const actions = computed<AutoTeleportAction[]>(() => [
+  {
+    action: action.value,
+    transaction,
+    details: {
+      isLoading: isLoading.value,
+      status: status.value,
+      isError: isError.value,
+      blockNumber: blockNumber.value,
+    },
+  },
+])
+
+const confirmButtonDisabled = computed(
+  () =>
+    hasOperationsDisabled(urlPrefix.value)
+    || offerStore.hasInvalidOfferPrices
+    || !autoTeleportButton.value?.isReady,
+)
+
+const confirmListingLabel = computed(() => {
+  if (!offerVisible(urlPrefix.value)) {
+    return $i18n.t('toast.unsupportedOperation')
+  }
+
+  if (!autoTeleportButton.value?.isReady) {
+    return $i18n.t('autoTeleport.checking')
+  }
+  return $i18n.t('transaction.offer')
+})
+
+const submitOffer = () => {
+  return transaction(getAction(offerStore.itemsInChain))
+}
+
+async function confirm({ autoteleport }: AutoTeleportActionButtonConfirmEvent) {
+  try {
+    clearTransaction()
+
+    autoTeleport.value = autoteleport
+    itemCount.value = offerStore.count
+    items.value = [...offerStore.itemsInChain]
+
+    if (!autoteleport) {
+      await submitOffer()
+    }
+
+    offerStore.clear()
+    closeMakingOfferModal()
+  }
+  catch (error) {
+    warningMessage(error)
+  }
+}
+
+const onClose = () => {
+  closeMakingOfferModal()
+}
+
+const handleSuccessModalClose = () => {
+  items.value = []
+}
+
+watch(
+  () => offerStore.count,
+  () => {
+    if (offerStore.count === 0) {
+      closeMakingOfferModal()
+    }
+  },
+)
+
+watch(
+  () => preferencesStore.makeOfferModalOpen,
+  (makeOfferModalOpen) => {
+    if (!makeOfferModalOpen) {
+      offerStore.clear()
+    }
+  },
+)
```

Comment:
```suggestion

```

Reply:
somehow I didn't push this change : )

Reply:
still there...

![CleanShot 2024-08-20 at 12 02 01@2x](https://github.com/user-attachments/assets/1039ad20-5fc9-4519-a0b5-f57221575112)